spec/system/new_reference_form_system_spec.rb

```diff
@@ -0,0 +1,98 @@
+require "system_helper"
+
+RSpec.describe "New Reference Form" do
+  let(:contributor) { create(:contributor) }
+  let(:admin) { create(:admin) }
+  let(:knownAuthor) { create(:author, name: "Known Author") }
+
+  before do
+    # these are for the homepage and are required for the redirection of not logged in users
+    create(:species, name: "Carcharhinus acronotus")
+    create(:species, name: "Prionace glauca")
+    create(:trait, name: "Lmat50")
+    create(:ocean, name: "Atlantic")
+  end
+
+  it "redirects to the main page if the user is not logged in" do
+    visit "/references/new"
+    expect(page).to have_content("If you wish to upload trait data, please make an account.")
+  end
+
+  it "creates a reference with no author" do
+    sign_in contributor
+    visit "/references/new"
+
+    expect(page).to have_content("New Reference")
+
+    fill_in "reference_name", with: "A new reference"
+    click_button "Create Reference"
+
+    expect(page).to have_content("A new reference")
+  end
+
+  it "creates a reference with a known author, with a dropdown that contains the known author name" do
+    sign_in contributor
+    visit "/references/new"
+
+    expect(page).to have_content("New Reference")
+
+    fill_in "reference_name", with: "A new reference"
+
+    fill_in "reference_authors_attributes_0_name", with: knownAuthor.name
```

Comment:
```suggestion
    fill_in "reference_authors_attributes_0_name", with: known_author.name[0..4]
```

We want the dropdown to show up if only a part of the name is entered.

Reply:
Ok - also added another step to click on the dropdown, so that the full name is populated - which made it so that the test passed.